js/client.js

```diff
@@ -2199,6 +2236,73 @@
 		}
 	});
 
+	Config.groups = Config.groups || {
+		'~': {
+			name: "Administrator (~)",
+			type: 2,
+			order: 1,
+		},
+		'#': {
+			name: "Room Owner (#)",
+			type: 2,
+			order: 2,
+		},
+		'&': {
+			name: "Leader (&amp;)",
+			type: 2,
+			order: 3,
+		},
+		'@': {
+			name: "Moderator (@)",
+			type: 1,
+			order: 4,
+		},
+		'%': {
+			name: "Driver (%)",
+			type: 1,
+			order: 5,
+		},
+		'*': {
+			name: "Bot (*)",
+			type: 0,
+			order: 6,
+		},
+		'\u2606': {
+			name: "Player (\u2606)",
+			type: 0,
+			order: 7,
+		},
+		'\u2605': {
+			name: "Player (\u2605)",
+			type: 0,
+			order: 8,
+		},
+		'+': {
+			name: "Voice (+)",
+			type: 0,
+			order: 9,
+		},
+		' ': {
+			type: 0,
+			order: 10,
+		},
+		'!': {
+			name: "<span style='color:#777777'>Muted (!)</span>",
+			type: 0,
+			order: 11,
```

Comment:
in parseGroups, i make it set ``Config.groups`` to ``{}`` first so the order won't cause any issues.  It will rewrite the punishgroups to use the neccesary order as it is getting parsed